scripts/lib/api/generateToc.test.ts

```diff
@@ -181,42 +179,55 @@ describe("generateToc", () => {
   test("TOC with grouped modules", () => {
     // This ordering is intentional.
     const topLevelEntries: TocGroupingEntry[] = [
-      { moduleId: "my_project", title: "API index", kind: "module" },
+      { moduleId: "my_quantum_project", title: "API index", kind: "module" },
       { name: "Group 2", kind: "section" },
       { name: "Group 1", kind: "section" },
+      {
+        moduleId: "my_quantum_project.another",
+        title: "dedicated module",
+        kind: "module",
+      },
```

Comment:
This is a new test because once we have https://github.com/Qiskit/documentation/issues/1249, we couldn't be confident that putting the API index page in the title came from our grouping code rather than the new `ensureIndexPage()`.